pages/add/add.jsx

```diff
@@ -140,21 +141,21 @@ export default React.createClass({
       </div>
     );
 
+    let getContent = function() {
```

Comment:
This should not be an inline function, but just a normal class function that returns function calls for the various bits of code.
```
  getContent() {
      if (user.loggedin) {
        return this.contentForLoggedInUser();
      }
      if (user.failedLogin) {
        return failurePrompt();
      }
      return anonymousContent();
    };
```
And then those three functions can, nice and contained, generate the content only when necessary, rather than generating all the content first, and then throwing away what we won't be using.